fableous-fe/src/components/canvas/Canvas.tsx

```diff
@@ -788,15 +790,23 @@ const Canvas = forwardRef<HTMLCanvasElement, CanvasProps>(
       // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [adjustCanvasSize, isShown]);
 
-    // initialize keyboard listener
+    // initialize undo shortcut listener
     useEffect(() => {
-      if (isShown) {
-        window.addEventListener("keydown", onKeyDown);
+      if (isShown && role !== ControllerRole.Hub) {
+        const undoListener = (event: KeyboardEvent) => {
+          if (event.key === "z" && (event.ctrlKey || event.metaKey)) {
+            placeUndo();
+          }
+        };
+        if (isShown) {
```

Comment:
conditional not needed here?

Reply:
i think its needed, we must prevent the hub from placing an undo

Reply:
this conditional is inside body of "isShown && role !== ControllerRole.Hub", if thats true, then if (isShown) will also always be true right?

Reply:
fixed